services/web-app/components/catalog/catalog.js

```diff
@@ -130,31 +271,33 @@ function Catalog({ collection, data, type, filter: defaultFilter = {}, glob = {}
     return totals
   })
 
+  // Set assets array that gets rendered as the filter, sort, and search changes. Avoid deep object
+  // equality comparison in the effect by using JSON.stringify.
   useEffect(() => {
-    setFilteredAssets(
-      assets
-        .sort(assetSortComparator(sort))
-        .filter((asset) => assetIsInFilter(asset, filter))
-        .filter((asset) => {
-          const { description = '', name = '' } = asset.content
-
-          if (search) {
-            return (
-              (name && name.toLowerCase().includes(search.toLowerCase())) ||
-              (description && description.toLowerCase().includes(search.toLowerCase()))
-            )
-          }
+    setFilteredAssets(getFilteredAssets(assets, filter, sort, search))
+    // Do not want to run this everytime the filter changes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [assets, JSON.stringify(filter), sort, search])
```

Comment:
using `fillter` in array deps instead now

Reply:
The `JSON.stringify` was probably from me, before this PR to check for object equality.

Reply:
`filter` object is probably getting reinstantiated somewhere, meaning it's a new object, meaning even if the values didn't change, the effect would still run. This was a hack (I admit) to prevent that.

Reply:
@jdharvey-ibm @francinelucca We might want to use something like https://github.com/streamich/react-use/blob/master/docs/useDeepCompareEffect.md for deep object comparisons in useEffect dependency arrays.

Reply:
@mattrosno ended up going for a `usePrevious` implementation to check if the filter actually changed before running the useEffect code , you can view that specific change here: https://github.com/carbon-design-system/carbon-platform/commit/b2b2e57af024850957cb62d3a9d5b6ccc6422808